pkg/kubelet/cm/container_manager_linux.go

```diff
@@ -649,8 +654,14 @@ func (cm *containerManagerImpl) Start(ctx context.Context, node *v1.Node,
 	return nil
 }
 
-func (cm *containerManagerImpl) GetPluginRegistrationHandler() cache.PluginHandler {
-	return cm.deviceManager.GetWatcherHandler()
+func (cm *containerManagerImpl) GetPluginRegistrationHandlers(getNode func() (*v1.Node, error)) map[string]cache.PluginHandler {
+	res := map[string]cache.PluginHandler{pluginwatcherapi.DevicePlugin: cm.deviceManager.GetWatcherHandler()}
+
+	if utilfeature.DefaultFeatureGate.Enabled(kubefeatures.DynamicResourceAllocation) {
+		res[pluginwatcherapi.DRAPlugin] = cache.PluginHandler(draplugin.NewRegistrationHandler(cm.kubeClient, getNode))
+	}
```

Comment:
Can we add a `GetWatcherHandler()` function to the DRA manager so that the call here is symmetrical with what we do for the device plugin? It also means we won't need the awkward `getNode` parameter to the `GetPluginRegistrationHandlers()` call.

Reply:
yes, can do. Do you suggest the dra manager will implement it's owne version of `getNode`? It makes sense

Reply:
hm, propagating `getNode` via NewManager's call looks as complicated.

I wonder if we actually want to use the version of getnode with the lister or we can re-implement it. I feel like we should just re-implement node getting function. Do you have objections?

Reply:
actually passed it to `Start` same as `ActivePodsFunc`.

Reply:
yes, this is what I envisioned -- just like we do for the device plugin already